Core/include/Acts/Navigation/DetectorNavigator.hpp

```diff
@@ -400,6 +428,11 @@ class DetectorNavigator {
 
     auto& nState = state.navigation;
 
+    auto volumeID = nState.currentVolume->geometryId().volume();
+
+    nState.externalSurfaceRange =
+        nState.options.externalSurfaces.equal_range(volumeID);
```

Comment:
I think the idea of external surface is to also put them into the candidates list which I don't see here

Reply:
When the `updateNavigationState` call is made by the `currentVolume`, we already have access to all the surfaces that are of interest to us within next several steps, including the external ones. Then we just have to properly treat the `BoundaryTolerance` in the `updateCandidates` to account for some surfaces being external. This way the external surfaces end up in the candidate list after the `updateNavigationState` call.